account-kit/java/signer-sdk/src/main/java/com/alchemy/aa/Stampler.java

```diff
@@ -0,0 +1,211 @@
+    byte[] aad = Bytes.concat(ephemeralPublicKeyUncompressed, tekPublicKeyBytes);
+
+    // Perform HPKE decryption
+    byte[] decryptedKey = tekManager.hpkeDecrypt(
+        ephemeralPublicKeyUncompressed,
+        ciphertext,
+        "turnkey_hpke".getBytes(),
+        aad
+    );
+
+    // Split out the privateKey/publicKey from the decrypted bytes
+    byte[][] keyPair = privateKeyToKeyPair(decryptedKey);
+    byte[] publicKeyBytes = keyPair[0];
+    byte[] privateKeyBytes = keyPair[1];
+
+    // Store them, as hex in lowercase
+    this.bundlePrivateKey = Utilities.bytesToHex(privateKeyBytes).toLowerCase(Locale.ROOT).toCharArray();
+    this.bundlePublicKey = Utilities.bytesToHex(publicKeyBytes).toLowerCase(Locale.ROOT).toCharArray();
+  }
+
+  /** Split out the privateKey/publicKey from the decrypted bytes
+   * @param privateKey
+   * @return byte struct key pairs.
+   */
+  private byte[][] privateKeyToKeyPair(byte[] privateKey) {
+    try {
+      // Create the EC private key
+      ECPrivateKey ecPrivateKey = EllipticCurves.getEcPrivateKey(EllipticCurves.CurveType.NIST_P256, privateKey);
+
+      // Use BouncyCastle to derive the public key
+      // Multiply base point G by the private scalar s
+      java.math.BigInteger s = ecPrivateKey.getS();
+      org.bouncycastle.jce.spec.ECParameterSpec bcSpec = ECNamedCurveTable.getParameterSpec("secp256r1");
+      ECPublicKeySpec pubSpec = new ECPublicKeySpec(bcSpec.getG().multiply(s).normalize(), bcSpec);
+      KeyFactory keyFactory = KeyFactory.getInstance("EC", BouncyCastleProvider.PROVIDER_NAME);
+
+      // Convert the result into Tink's EcPublicKey format
+      ECPublicKey ecPublicKey = EllipticCurves.getEcPublicKey(EllipticCurves.CurveType.NIST_P256,
+          EllipticCurves.PointFormatType.COMPRESSED, keyFactory.generatePublic(pubSpec).getEncoded());
+
+      // Validate they match
+      EllipticCurves.validatePublicKey(ecPublicKey, ecPrivateKey);
+
+      // Compress the public key to match Turnkey's expectation
+      byte[] compressedPublicKey = ecPublicKey.getEncoded();
+
+      // Return both
+      return new byte[][] {compressedPublicKey, privateKey};
+
+    } catch (Exception e) {
+      throw new RuntimeException("Error deriving public key from private key", e);
+    }
+  }
+
+  /**
+   * Signs the given payload using the stored private key and returns a Stamp.
+   * @param payload payload to sign
+   * @return signed stamp
+   * @throws GeneralSecurityException iif the private is malformed
+   */
+  public Stamp stamp(String payload) throws GeneralSecurityException {
+    if (this.bundlePrivateKey == null || this.bundlePublicKey == null) {
+      throw new NoInjectedBundleException();
+    }
+    // Retrieve the private key from shared prefs
+    String signingKeyHex = String.valueOf(this.bundlePrivateKey);
+    String publicSigningKeyHex = String.valueOf(this.bundlePublicKey);
+
+
+
+    byte[] privateKeyBytes = Utilities.HexToBytes(signingKeyHex);
```

Comment:
ah yea here is what I was thinking we could skip if the stamper just stores the keys as bytes